tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h

```diff
@@ -1928,6 +1928,126 @@ inline void Div(const float* input1_data, const Dims<4>& input1_dims,
+          output_data[Offset(output_dims, c, x, y, b)] =
+              ActivationFunctionWithMinMax(
+                  input1_data[SubscriptToIndex(desc1, c, x, y, b)] /
+                      input2_data[SubscriptToIndex(desc2, c, x, y, b)],
+                  output_activation_min, output_activation_max);
+        }
+      }
+    }
+  }
+}
+
+// legacy, for compatibility with old checked-in code
+template <FusedActivationFunctionType Ac, typename T>
+void BroadcastDiv(const T* input1_data, const Dims<4>& input1_dims,
+                  const T* input2_data, const Dims<4>& input2_dims,
+                  T* output_data, const Dims<4>& output_dims) {
+  T output_activation_min, output_activation_max;
+  GetActivationMinMax(Ac, &output_activation_min, &output_activation_max);
+
+  BroadcastDiv(input1_data, input1_dims, input2_data, input2_dims,
+               output_activation_min, output_activation_max, output_data,
+               output_dims);
+}
+
+inline void BroadcastDiv(const uint8* input1_data, const Dims<4>& input1_dims,
+                         int32 input1_offset, const uint8* input2_data,
+                         const Dims<4>& input2_dims, int32 input2_offset,
+                         int32 output_offset, int32 output_multiplier,
+                         int output_shift, int32 output_activation_min,
+                         int32 output_activation_max, uint8* output_data,
+                         const Dims<4>& output_dims) {
+  gemmlowp::ScopedProfilingLabel label("BroadcastDiv/8bit");
+
+  NdArrayDesc<4> desc1;
+  NdArrayDesc<4> desc2;
+  NdArrayDescsForElementwiseBroadcast(input1_dims, input2_dims, &desc1, &desc2);
+
+  // In Tensorflow, the dimensions are canonically named (batch_number, row,
+  // col, channel), with extents (batches, height, width, depth), with the
+  // trailing dimension changing most rapidly (channels has the smallest stride,
+  // typically 1 element).
+  //
+  // In generated C code, we store arrays with the dimensions reversed. The
+  // first dimension has smallest stride.
+  //
+  // We name our variables by their Tensorflow convention, but generate C code
+  // nesting loops such that the innermost loop has the smallest stride for the
+  // best cache behavior.
+  for (int b = 0; b < ArraySize(output_dims, 3); ++b) {
+    for (int y = 0; y < ArraySize(output_dims, 2); ++y) {
+      for (int x = 0; x < ArraySize(output_dims, 1); ++x) {
+        for (int c = 0; c < ArraySize(output_dims, 0); ++c) {
+          const int32 input1_val =
+              input1_offset + input1_data[SubscriptToIndex(desc1, c, x, y, b)];
+          const int32 input2_val =
+              input2_offset + input2_data[SubscriptToIndex(desc2, c, x, y, b)];
+          const int32 unclamped_result =
+              output_offset +
+              MultiplyByQuantizedMultiplierSmallerThanOne(
+                  input1_val / input2_val, output_multiplier, output_shift);
```

Comment:
There's an accuracy issue here as the integer division will truncate a lot of values.

Leaving that aside for a second, the calculation doesn't seem correct either (quantization is hard!). The quantized multiplier is s1*s2/sout so the result here is
   s1*s2/sout*i1/i2
but it should be  s1/s2*i1/i2/sout, right?

Again, quantization is hard, so I'll be OK if you prefer to  leave quantization for a subsequent PR.

One solution is to calculate a quantized multiplier that is always greater than 1:
  qm = (s1> s2*sout) ? qmul(s1/s2/sout) : qmul(s2*sout/s1)
and pass a hint as to whether the numerator or the denominator needs scaling
  return  denominator_needs_scaling ? q1 / (q2 * qm) : (q1 * qm) / q2;

Reply:
Please can you provide more details or link to docs so I can understand quantized ops better.

Reply:
The quantization scheme used in TF Lite is similar to gemmlowp's: https://github.com/google/gemmlowp/blob/master/doc/quantization.md

Maybe we can leave the quantization for a separate PR? I'd be happy to approve the other changes, and then we can have someone with more quantization experience review the other PR.